hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/TestReplicationManager.java

```diff
@@ -897,6 +1061,63 @@ public void testOverReplicatedAndPolicyUnSatisfiedAndDeleted() throws
     Thread.sleep(100L);
     Assert.assertEquals(currentDeleteCommandCount + 2, datanodeCommandHandler
         .getInvocationCount(SCMCommandProto.Type.deleteContainerCommand));
+    Assert.assertEquals(currentDeleteCommandCount + 2,
+        replicationManager.getMetrics().getNumDeletionCmdsSent());
+    Assert.assertEquals(1, replicationManager.getInflightDeletion().size());
+    Assert.assertEquals(1, replicationManager.getMetrics()
+        .getInflightDeletion());
+  }
+
+  @Test
+  public void testReplicateCommandTimeout() throws
+      SCMException, ContainerNotFoundException, InterruptedException {
+    replicationManager.stop();
+    ReplicationManagerConfiguration newConf =
+        new ReplicationManagerConfiguration();
+    // Set a short timeout
+    newConf.setEventTimeout(Duration.ofSeconds(1));
+    createReplicationManager(newConf);
+    final ContainerInfo container = createContainer(LifeCycleState.CLOSED);
+    addReplica(container, new NodeStatus(IN_SERVICE, HEALTHY), CLOSED);
+    addReplica(container, new NodeStatus(IN_SERVICE, HEALTHY), CLOSED);
+    assertReplicaScheduled(1);
+
+    // Make a timeout
+    Thread.sleep(2000);
+
+    replicationManager.processContainersNow();
+    // Wait for EventQueue to call the event handler
+    Thread.sleep(100L);
+
+    Assert.assertEquals(1, replicationManager.getMetrics()
+        .getNumReplicationCmdsTimeout());
+  }
+
+  @Test
+  public void testDeleteCommandTimeout() throws
+      SCMException, ContainerNotFoundException, InterruptedException {
+    replicationManager.stop();
+    ReplicationManagerConfiguration newConf =
+        new ReplicationManagerConfiguration();
+    // Set a short timeout
+    newConf.setEventTimeout(Duration.ofSeconds(1));
+    createReplicationManager(newConf);
+    final ContainerInfo container = createContainer(LifeCycleState.CLOSED);
+    addReplica(container, new NodeStatus(IN_SERVICE, HEALTHY), CLOSED);
+    addReplica(container, new NodeStatus(IN_SERVICE, HEALTHY), CLOSED);
+    addReplica(container, new NodeStatus(IN_SERVICE, HEALTHY), CLOSED);
+    addReplica(container, new NodeStatus(IN_SERVICE, HEALTHY), CLOSED);
+    assertDeleteScheduled(1);
+
+    // Make a timeout
+    Thread.sleep(2000);
```

Comment:
I'd like to avoid these sleeps in the tests. I have created #2425 to add a Clock into Replication Manager. If we can get that reviewed and committed then we can rebase this patch and remove the timeouts.

Reply:
Ah, that's great, I see that it is merged now and I'll do a rebase soon, thanks~

Reply:
Oh, I noticed that there is another improvement pr https://github.com/apache/ozone/pull/2429, I shall wait for it merged and rebase on that.